presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PruneUnreferencedOutputs.java

```diff
@@ -460,6 +461,7 @@ public PlanNode visitMarkDistinct(MarkDistinctNode node, RewriteContext<Set<Symb
 
             ImmutableSet.Builder<Symbol> expectedInputs = ImmutableSet.<Symbol>builder()
                     .addAll(node.getDistinctSymbols())
+                    .addAll(node.getOrderBySymbols())
```

Comment:
I was told to break change to AggregationNode and MarkDistinctNode to a separate commit so it's easier to see what are the changes in optimizer. If I merge changes using these functions and changes adding them together I'm back to the beginning. So do we want one commit or two? If two, could you please make it clearer what exactly goes to each?